### PR TITLE
[MM-41189] Remove "Invite to team" feature flag 

### PR DESCRIPTION
**Pull request by JulienTant**
#### Summary
The result of the AB testing led us to think that what was in place is better than the new variant. This PR removes the feature flag and the new variant.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-38799

#### Related Pull Requests
- Has server changes
https://github.com/mattermost/mattermost-server/pull/19426

#### Screenshots
N/A

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by neallred — APPROVED** [review]
lgtm!

File: components/invitation_modal/invite_as.tsx [hunk 1]
```diff
@@ -25,102 +20,55 @@ export type Props = {
...
+                            value: InviteType.MEMBER,
+                            testId: 'inviteMembersLink',
+                        },
+                        {
+                            key: (
+                                <span className='InviteAs__label'>
                                     <FormattedMessage
-                                        id='invite_modal.choose_guest_b'
-                                        defaultMessage='limited to select channels and teams'
+                                        id='invite_modal.choose_guest_a'
```
Comment (marianunez) [inline components/invitation_modal/invite_as.tsx:54]: nit: we can drop the `a` reference of the text

File: components/invitation_modal/invite_as.tsx [hunk 2]
```diff
@@ -25,102 +20,55 @@ export type Props = {
...
+                                        defaultMessage='Guest'
                                     />
+                                    <span className='InviteAs__label--parenthetical'>
+                                        {' - '}
+                                        <FormattedMessage
+                                            id='invite_modal.choose_guest_b'
```
Comment (marianunez) [inline components/invitation_modal/invite_as.tsx:60]: nit: Same here

**Review by marianunez — APPROVED** [review]
LGTM, minor nit below

File: components/invitation_modal/invite_as.tsx (hunk 1 above)
Comment (JulienTant) [reply components/invitation_modal/invite_as.tsx:54]: they exist as a way to differentiate two phrases in the same area and are not related to the a/b testing.

Comment (marianunez) [reply components/invitation_modal/invite_as.tsx:54]: Gotcha, that's for clarifying!

**Review by stevemudie — APPROVED** [review]
Looks good and worked as expected.

**Pull request merged**